apps/wing/src/analytics/storage.ts

```diff
@@ -0,0 +1,96 @@
+import { readFileSync, writeFileSync, mkdirSync, existsSync } from "fs";
+import { AnalyticEvent } from "./event";
+import { randomBytes } from "crypto";
+import {v4 as uuidv4 } from "uuid";
+import path from "path";
+import * as os from "os";
+
+const WING_HOME_DIR = path.join(os.homedir(), ".wing")
+const WING_ANALYTICS_CONFIG_FILE = path.join(WING_HOME_DIR, 'wing-analytics-config.json');
+
+interface AnalyticsConfig {
+  anonymousId: string;
+  disclaimerDisplayed?: boolean;
+  optOut?: boolean;
+}
+
+/**
+ * Stores an analytic event to disk for later export
+ * 
+ * @param event the event to store
+ * @returns string the file path of the stored analytic
+ */
+export function storeAnalyticEvent(event: AnalyticEvent): string | undefined {
+  try {
+    const eventId = uuidv4();
+    const tmpdir = path.join(os.tmpdir(), "wing-analytics");
+    
+    if (!existsSync(tmpdir)) {
+      mkdirSync(tmpdir);
+    }
+
+    const analyticReportFile = path.join(tmpdir, `${eventId}.json`);
+    
+    const anonymousId = getAnonymousId();
+  
+    // attach timestamp and anonymousId to event
+    event.timestamp = event.timestamp ?? new Date().toISOString();
+    event.anonymousId = anonymousId;
+  
+    saveAnalyticsReport(analyticReportFile, event);
+  
+    return analyticReportFile;
+
+  } catch (error) {
+    return undefined;
+  }
+
+}
+
+export function getAnonymousId(): string {
+  let config = loadAnalyticsConfig();
+  return config.anonymousId;
+}
+
+export function loadAnalyticsConfig(): AnalyticsConfig {
+  try {
+    const fileContents = readFileSync(WING_ANALYTICS_CONFIG_FILE, 'utf-8');
+    return JSON.parse(fileContents) as AnalyticsConfig;
+  } catch (error: any) {
+    // Possible issue reading analytics config file, first determine if file exists yet
+
+    if (error.code === 'EACCES') {
+      throw new Error("Not able to access analytics config file due to permissions");
+    }
+
+    const analyticsConfig: AnalyticsConfig = {
+      anonymousId: randomBytes(16).toString('hex'),
+      optOut: false,
+    }
+
+    saveAnalyticsConfig(analyticsConfig);
+
+    return analyticsConfig;
+  }
+}
```

Comment:
@hasanaburayyan , 

I was thinking that in CI we will not have this value, the result is that in every CI run the system will generate a new `anonymousId` 

One option I have is to take the git repo and run a hash function on it (e.g. md5) and using the hashed versions  as an anonymousId in the case of CI 

I this case we can't determine the real github repo, (unless we guess it and run the same hash function) 

It might be also nice to add this anonymouRepoId in all the events, we can use it to understand correlation  between anonymouId and anonymouRepoId  without any privacy issues

Reply:
actually anonymouRepoId should be inside `properties` and should be called : `anonymou_repo_id`

Reply:
@ainvoner - notice that the console should also have the `anonymou_repo_id` property

Reply:
yes, I am full of typos

Reply:
Also @ekeren I believe we have to keep anonymousID even if it was in CI this is because its a required value in segment. Though if its in CI I can just make the value say "see anonymous_repo_id"